diff --git a/configure b/configure
index 7acb4a8..a85c999 100755
--- a/configure
+++ b/configure
@@ -655,7 +655,6 @@ infodir
 docdir
 oldincludedir
 includedir
-runstatedir
 localstatedir
 sharedstatedir
 sysconfdir
@@ -729,7 +728,6 @@ datadir='${datarootdir}'
 sysconfdir='${prefix}/etc'
 sharedstatedir='${prefix}/com'
 localstatedir='${prefix}/var'
-runstatedir='${localstatedir}/run'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
 docdir='${datarootdir}/doc/${PACKAGE}'
@@ -982,15 +980,6 @@ do
   | -silent | --silent | --silen | --sile | --sil)
     silent=yes ;;
 
-  -runstatedir | --runstatedir | --runstatedi | --runstated \
-  | --runstate | --runstat | --runsta | --runst | --runs \
-  | --run | --ru | --r)
-    ac_prev=runstatedir ;;
-  -runstatedir=* | --runstatedir=* | --runstatedi=* | --runstated=* \
-  | --runstate=* | --runstat=* | --runsta=* | --runst=* | --runs=* \
-  | --run=* | --ru=* | --r=*)
-    runstatedir=$ac_optarg ;;
-
   -sbindir | --sbindir | --sbindi | --sbind | --sbin | --sbi | --sb)
     ac_prev=sbindir ;;
   -sbindir=* | --sbindir=* | --sbindi=* | --sbind=* | --sbin=* \
@@ -1128,7 +1117,7 @@ fi
 for ac_var in	exec_prefix prefix bindir sbindir libexecdir datarootdir \
 		datadir sysconfdir sharedstatedir localstatedir includedir \
 		oldincludedir docdir infodir htmldir dvidir pdfdir psdir \
-		libdir localedir mandir runstatedir
+		libdir localedir mandir
 do
   eval ac_val=\$$ac_var
   # Remove trailing slashes.
@@ -1281,7 +1270,6 @@ Fine tuning of the installation directories:
   --sysconfdir=DIR        read-only single-machine data [PREFIX/etc]
   --sharedstatedir=DIR    modifiable architecture-independent data [PREFIX/com]
   --localstatedir=DIR     modifiable single-machine data [PREFIX/var]
-  --runstatedir=DIR       modifiable per-process data [LOCALSTATEDIR/run]
   --libdir=DIR            object code libraries [EPREFIX/lib]
   --includedir=DIR        C header files [PREFIX/include]
   --oldincludedir=DIR     C header files for non-gcc [/usr/include]
@@ -1314,8 +1302,8 @@ Optional Features:
 Optional Packages:
   --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
   --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
-  --with-ical-include=DIR  include path for ical headers
-  --with-ical-library=DIR  library path for ical libraries
+  --with-ical-include=DIR include path for ical headers
+  --with-ical-library=DIR library path for ical libraries
 
 Some influential environment variables:
   CC          C compiler command
@@ -2052,7 +2040,7 @@ additional_lib_dir=
 
 # Check whether --with-ical-include was given.
 if test "${with_ical_include+set}" = set; then :
-  withval=$with_ical_include; additional_include_dir+=-I"$withval"
+  withval=$with_ical_include; additional_include_dir=-I"$withval"
 fi
 
 
diff --git a/configure.ac b/configure.ac
index 431ac84..45b1075 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,8 +13,12 @@ old_ldflags=$LDFLAGS
 additional_include_dir=
 additional_lib_dir=
 
-AC_ARG_WITH(ical-include, [  --with-ical-include=DIR  include path for ical headers], additional_include_dir+=-I"$withval", )
-AC_ARG_WITH(ical-library, [  --with-ical-library=DIR  library path for ical libraries], additional_lib_dir=-L"$withval", )
+AC_ARG_WITH(ical-include,
+	[AS_HELP_STRING([--with-ical-include=DIR],[include path for ical headers])],
+	[additional_include_dir=-I"$withval"], )
+AC_ARG_WITH(ical-library,
+	[AS_HELP_STRING([--with-ical-library=DIR], [library path for ical libraries])],
+	[additional_lib_dir=-L"$withval"], )
 CPPFLAGS="$CPPFLAGS $additional_include_dir"
 LDFLAGS="$LDFLAGS $additional_lib_dir"
 AC_CHECK_HEADERS([libical/ical.h], ,[AC_CHECK_HEADERS([ical.h], ,[AC_MSG_ERROR(ical.h not found. You must have libical >= 0.27 installed.)])])
